### PR TITLE
Add new quest to ask if a place is suitable for vegans or vegetarians

### PR DESCRIPTION
**Pull request by ENT8R**
This PR adds a new quest to determine if a place is suitable for vegans or vegetarians. (See #479). It looks like this: 

The quest doesn't have an icon yet too, but maybe the [fork and knife emoji from twemoji](https://commons.wikimedia.org/wiki/File:Twemoji2_1f374.svg) or the [fork, knife and spoon emoji from twemoji](https://commons.wikimedia.org/wiki/File:Twemoji2_1f37d.svg) are a possible solution...

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		}
+		else
+		{
+			switch (answer)
+			{
+				case AddDietTypeForm.VEGETARIAN:
+					changes.add("diet:vegetarian", "yes");
+					break;
+				case AddDietTypeForm.VEGAN:
+					changes.add("diet:vegan", "yes");
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java]: This should be aditionally:
`changes.add("diet:vegetarian", "yes");`

Comment (ENT8R) [inline app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java]: Yes, you're right! I added it now...

Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java]: Do you have an idea with the vegan:only code path? 
Should we have 
`changes.add("diet:vegetarian", "only");` and `changes.add("diet:vegetarian", "yes");` 
looks strange.

Comment (ENT8R) [inline app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java]: What do you mean?

File: app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	public void applyAnswerTo(Bundle bundle, StringMapChangesBuilder changes)
+	{
+		String other = bundle.getString(AddDietTypeForm.OTHER_ANSWER);
+		String answer = bundle.getString(AddDietTypeForm.ANSWER);
+		if (other != null)
+		{
+			switch (other)
+			{
+				case AddDietTypeForm.ONLY_VEGAN:
+					changes.add("diet:vegan", "only");
```
Comment (HolgerJeromin) [inline app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java]: Could probably be
`changes.add("diet:vegetarian", "only");` 
additional. But i found no object in europe with an vegan=only tag and an vegetarian tag. Overpass API found about 600 with vegan=only.

File: app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java (hunk 1 above)
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/diet_type/AddDietType.java]: ehm, wait a moment here! `diet:vegan` implies `diet:vegetarian`, so this should **not** be tagged additionally!

**Comment by ENT8R** [comment]
I will close this and make a new pull request.

**Comment by rugk** [comment]
> Then, only for places with vegetation=yes, ask if it is also fit for vegans.

Yeah for "vegetation=yes"… :deciduous_tree:  This made my day. :laughing:

**Pull request closed without merging**